### PR TITLE
Add missing client apis

### PR DESCRIPTION
**Pull request by apanicker-nflx**

### CODE REVIEW

File: jersey/src/main/java/com/netflix/conductor/server/resources/TaskResource.java [hunk 1]
```diff
@@ -209,7 +209,7 @@ public void remvoeTaskFromQueue(@PathParam("taskType") String taskType, @PathPar
 		Set<Entry<String, Long>> sorted = new TreeSet<>(Comparator.comparing(Entry::getKey));
 		sorted.addAll(entries);
 		LinkedHashMap<String, Long> sortedMap = new LinkedHashMap<>();
-		sorted.stream().forEach(e -> sortedMap.put(e.getKey(), e.getValue()));
+		sorted.forEach(e -> sortedMap.put(e.getKey(), e.getValue()));
```
Comment (pctreddy) [inline jersey/src/main/java/com/netflix/conductor/server/resources/TaskResource.java]: All of this code can be changed to:
```java
return queues.queuesDetail().entrySet().stream()
				.sorted(Comparator.comparing(Entry::getKey))
				.collect(Collectors.toMap(Entry::getKey, Entry::getValue, (v1, v2) -> v1, LinkedHashMap::new));

File: redis-persistence/src/main/java/com/netflix/conductor/dao/index/ElasticSearchDAO.java [hunk 2]
```diff
@@ -106,6 +106,8 @@
 	private static final String MSG_DOC_TYPE = "message";
 	
 	private static final String className = ElasticSearchDAO.class.getSimpleName();
+
+	private static final int RETRY_COUNT = 3;
```
Comment (pctreddy) [inline redis-persistence/src/main/java/com/netflix/conductor/dao/index/ElasticSearchDAO.java]: Same as above consider making this retry count configurable.

File: client/src/main/java/com/netflix/conductor/client/http/MetadataClient.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        Preconditions.checkArgument(StringUtils.isNotBlank(name), "name cannot be blank");
+        return getForEntity("metadata/workflow/{name}", new Object[]{"version", version}, WorkflowDef.class, name);
+    }
+
+    /**
+     * Retrieve all workflow definitions
+     *
+     * @return List of all workflow definitions registered with the server
+     */
+    public List<WorkflowDef> getAllWorkflowDefs() {
```
[the remark this answers is not in the document]
Comment (apanicker-nflx) [reply client/src/main/java/com/netflix/conductor/client/http/MetadataClient.java]: Search API cannot search for blueprints, only workflow/task instances are indexed and available for searching.

File: client/src/main/java/com/netflix/conductor/client/http/MetadataClient.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        Preconditions.checkNotNull(taskDef, "Task definition cannot be null");
+        put("metadata/taskdefs", null, taskDef);
+    }
+
+    /**
+     * Retrieve all task definitions
+     *
+     * @return List of all the task definitions registered with the server
+     */
+    public List<TaskDef> getAllTaskDefs() {
```
[the remark this answers is not in the document]
Comment (apanicker-nflx) [reply client/src/main/java/com/netflix/conductor/client/http/MetadataClient.java]: see above comment.

File: es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchDAO.java [hunk 5]
```diff
@@ -108,6 +107,8 @@
 	private static final String MSG_DOC_TYPE = "message";
 	
 	private static final String className = ElasticSearchDAO.class.getSimpleName();
+
+	private static final int RETRY_COUNT = 3;
```
[the remark this answers is not in the document]
Comment (apanicker-nflx) [reply es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchDAO.java]: This is internal implementation detail and would not benefit from making configurable.

File: jersey/src/main/java/com/netflix/conductor/server/resources/TaskResource.java (hunk 1 above)
Comment (apanicker-nflx) [reply jersey/src/main/java/com/netflix/conductor/server/resources/TaskResource.java]: done

**Pull request merged**